### PR TITLE
Home work 8

### PR DESCRIPTION
**Pull request by SlavikSavenok**

### CODE REVIEW

File: src/homeWork_8_1/Converter.java [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package homeWork_8_1;
+
+public interface Converter {
+
+    double getCovert(double temp);
```
Comment (dmitars) [inline src/homeWork_8_1/Converter.java:5]: Possibly, convert?

File: src/homeWork_8_1/BaseConverter.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+        System.out.print("Enter the value to convert: ");
+        double temp = scanner.nextDouble();
+
+
+
+        System.out.println("The temperature in Celsius is: " + celsius.getCovert(temp));
+        System.out.println("The temperature in Fahrenheit is: " + fahrenheit.getCovert(temp));
+        System.out.println("The temperature in Kelvin is: " + kelvin.getCovert(temp));
+
```
Comment (dmitars) [inline src/homeWork_8_1/BaseConverter.java:20]: Where is static method, that can convert degrees to degrees for this task?

File: src/homeWork_8_3/AnimalEnum.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
+package homeWork_8_3;
+
+public enum AnimalEnum implements Action {
+    UNICORN(),
+    PHOENIX(),
+    KRAKEN(),
+    PEGASUS(),
+    CERBERUS(),
+    LEVIATHAN();
```
Comment (dmitars) [inline src/homeWork_8_3/AnimalEnum.java:9]: We don't need to add default constructor call for constants in enum

File: src/homeWork_8_3/AnimalEnum.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    boolean canSwim;
+
```
Comment (dmitars) [inline src/homeWork_8_3/AnimalEnum.java:11]: Why do we need this variable?

File: src/homeWork_8_3/AnimalEnum.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    @Override
+    public String getFly() {
+        return name() + " flying";
+    }
+
+    @Override
+    public String getSay() {
+        return name() + " saying";
```
Comment (dmitars) [inline src/homeWork_8_3/AnimalEnum.java:33]: getEat -> eat, getSwim -> swim and so on

File: src/homeWork_8_3/AnimalEnum.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        return name() + " eating";
+    }
+
+    @Override
+    public String getSwim() {
+        return name() + " swimming";
+    }
+
+    @Override
+    public String getFly() {
```
Comment (dmitars) [inline src/homeWork_8_3/AnimalEnum.java:27]: Animals, that can't fly or swim, even shouldn't have this method. To add animals as enum worse then work with them as different objects

File: src/homeWork_8_3/HomeWork_8_3.java [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+            return AnimalEnum.UNICORN;
+        if (selectFlag == 1)
+            return AnimalEnum.CERBERUS;
+        if (selectFlag == 2)
+            return AnimalEnum.KRAKEN;
+        if (selectFlag == 3)
+            return AnimalEnum.PEGASUS;
+        if (selectFlag == 4)
+            return LEVIATHAN;
+        else return AnimalEnum.PHOENIX;
```
Comment (dmitars) [inline src/homeWork_8_3/HomeWork_8_3.java:68]: return AnimalEnum.values()[selectFlag]

File: src/homeWork_8_3/AnimalEnum.java (hunk 4 above)
Comment (SlavikSavenok) [reply src/homeWork_8_3/AnimalEnum.java:11]: I tried to solve it using boolean (for example, it can swim). Forgot to delete

**Pull request merged**